### PR TITLE
LPS-185604 Add component to handle action executions

### PR DESCRIPTION
**Pull request by sandrodw3**
⚠️ Only LPS-185604 commits need to be reviewed

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by jkappler** [comment]
@ruben-pulido, looks good after what we talked about ❤️

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/render_layout_structure/js/InfoItemActionHandler.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+function triggerAction(trigger, executeInfoItemActionURL) {
+	const {
+		lfrClassNameId: classNameId,
+		lfrClassPk: classPK,
+		lfrFieldId: fieldId,
+	} = trigger.dataset;
+
+	if (!fieldId) {
```
Comment (victorg1991) [inline modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/render_layout_structure/js/InfoItemActionHandler.js:59]: Only fieldId is optional?

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/render_layout_structure/js/InfoItemActionHandler.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+		.then((response) => response.json())
+		.then(({error}) => {
+			trigger.classList.remove('disabled');
+			trigger.removeChild(loadingIndicator);
+
+			if (error) {
+				openResultToast('error', error);
+			}
+			else {
+				openResultToast('success');
```
Comment (victorg1991) [inline modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/render_layout_structure/js/InfoItemActionHandler.js:85]: Maybe we justo do here TOAST_DATA.success so we don't have to use any string here

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/render_layout_structure/js/InfoItemActionHandler.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+		lfrFieldId: fieldId,
+	} = trigger.dataset;
+
+	if (!fieldId) {
+		return;
+	}
+
+	const loadingIndicator = getLoadingIndicator();
+
+	trigger.classList.add('disabled');
```
Comment (victorg1991) [inline modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/render_layout_structure/js/InfoItemActionHandler.js:65]: Buttons with `disabled` class are still focusable and you can trigger the action again right? Probably we need to set the disabled attribute here too

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/render_layout_structure/js/InfoItemActionHandler.js (hunk 1 above)
Comment (sandrodw3) [reply modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/render_layout_structure/js/InfoItemActionHandler.js]: After discussing it with Ruben, we have agreed on removing the if for now, so it throws the default error when some field is missing, and for the second story, we are going to ask Design and PM what we want to do in the case the fragment is not mapped to any action. Thanks!

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/render_layout_structure/js/InfoItemActionHandler.js (hunk 2 above)
Comment (sandrodw3) [reply modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/render_layout_structure/js/InfoItemActionHandler.js]: I like the idea! Changed it

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/render_layout_structure/js/InfoItemActionHandler.js (hunk 3 above)
Comment (sandrodw3) [reply modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/render_layout_structure/js/InfoItemActionHandler.js:61]: Done! thanks 😄

**Comment by sandrodw3** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Comment by jkappler** [comment]
ci:forward

**Comment by sandrodw3** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Comment by ruben-pulido** [comment]
https://github.com/liferay-echo/liferay-portal/pull/12409

**Comment by ruben-pulido** [comment]
ci:forward

**Comment by ruben-pulido** [comment]
Good [finding](https://github.com/liferay-echo/liferay-portal/pull/12383/commits/4de327a7d215fe7a0f42496bd4b8bfe47d819664) @sandrodw3 and @victorg1991!

**Comment by ruben-pulido** [comment]
ci:forward:force

**Comment by ruben-pulido** [comment]
ci:forward:force

**Pull request closed without merging**